assets/js/utility-animation.js

```diff
@@ -0,0 +1,48 @@
+import State from './state';
+
+class UtilityAnimation {
+  static rootSelector = '.utility-animation';
+  static instances = [];
+
+  constructor() {
+    this.count = 1;
+    this.currentElement = document.querySelectorAll('[data-utility-animation-step="1"]');
+    this.initialize();
+  }
+
+  handleAnimationEnd() {
+    this.count++;
+    const searchQuery = `[data-utility-animation-step="${this.count}"]`;
+
+    this.toggleState(this.currentElement);
+    this.currentElement = document.querySelectorAll(searchQuery);
+
+    if (this.currentElement !== null && this.currentElement.length > 0) {
+      this.toggleState(this.currentElement);
+    }
+  }
+
+  toggleState(element) {
+    if (element.length > 1) {
+      element.forEach((e) => {
+        e.classList.toggle(State.IS_STARTING);
+      });
+    } else {
+      element[0]?.classList.toggle(State.IS_STARTING);
+    }
+  }
+
+  initialize() {
+    this.toggleState(this.currentElement);
```

Comment:
Hier toggle ich den State und in handleAnimationEnd rufe ich doch dann toggleState nochmal auf, mit dem selben currentElement oder?